core/src/main/java/org/frankframework/senders/ParallelSenderExecutor.java

```diff
@@ -15,61 +15,57 @@
 */
 package org.frankframework.senders;
 
-import org.apache.logging.log4j.Logger;
+import java.util.concurrent.Phaser;
+
+import io.micrometer.core.instrument.DistributionSummary;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
 import org.frankframework.core.ISender;
 import org.frankframework.core.PipeLineSession;
 import org.frankframework.core.RequestReplyExecutor;
+import org.frankframework.receivers.ResourceLimiter;
 import org.frankframework.stream.Message;
-import org.frankframework.util.Guard;
-import org.frankframework.util.LogUtil;
-import org.frankframework.util.Semaphore;
-
-import io.micrometer.core.instrument.DistributionSummary;
-import lombok.Getter;
 
+@Log4j2
 public class ParallelSenderExecutor extends RequestReplyExecutor {
-	private Logger log = LogUtil.getLogger(this);
-	private ISender sender;
-	@Getter private PipeLineSession session;
-	private Semaphore semaphore; // supports to limit the number of threads processing in parallel, may be null
-	private Guard guard;         // supports to wait for all threads to have ended
-	private DistributionSummary summary;
+	private final ISender sender;
+	@Getter private final PipeLineSession session;
+	@Setter private ResourceLimiter limiter; // support limiting the number of threads processing in parallel
```

Comment:
Misschien is `threadLimiter` of zo een betere naam voor deze variabele, dan is sneller duidelijk wat wordt gelimiteerd.

Reply:
Dat staat er ook al achter. Ik vind het zo ook wel prima eigenlijk.